### DISCUSSION TITLE
Rendering a Cube with 8 vertex?

### DISCUSSION THREAD

**Discussion by paulojsmmm** [google/filament#4355] [Q&A]
On SceneKit (iOS) I have an object with 400 vertex and 2166 indices. To draw a Cube there I only need 8 vertex and 24 indices.
Can I do the same on Filament? I developing for Android right now and I checked the Cube example: they use 24 vertex and 24 indices to do it. Do I need to transform my 400 vertex into a 2166 one?

Also: Is it possible to let the User to (freely) rotate the screen?

Thank you

**Comment by romainguy** [comment]
The sample uses 24 vertices so that each face has different normals to get the intended lighting. There is however nothing preventing you from using 8 vertices and 24 indices.

And yes the user can rotate the screen, this has nothing to do with Filament.

**Reply by paulojsmmm** [reply]
Rotate the screen I mean: rotate the cube (change camera zoom etc) with his finger
